### PR TITLE
NIAD-776 Receive EHR Extract Request, update database, and create further tasks

### PR DESCRIPTION
**Pull request by mattd-kainos**

### CODE REVIEW

**Review by Broom94 — CHANGES_REQUESTED** [review]
Really great job refactoring the code base, looks great! Few minor comments really nice job with this!

File: service/src/main/java/uk/nhs/adaptors/gp2gp/common/task/TaskDefinition.java [hunk 1]
```diff
@@ -1,16 +1,16 @@
 package uk.nhs.adaptors.gp2gp.common.task;
 
+import lombok.EqualsAndHashCode;
 import lombok.Getter;
+import lombok.experimental.SuperBuilder;
 
 @Getter
+@SuperBuilder
```
[the remark this answers is not in the document]
Comment (mattd-kainos) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/common/task/TaskDefinition.java:8]: no, its super!

File: service/src/test/java/uk/nhs/adaptors/gp2gp/common/task/TaskConsumerTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    @InjectMocks
+    private TaskConsumer taskConsumer;
+
+    @Mock
+    private Message message;
+
+    @Test
+    @SneakyThrows
+    public void When_TaskHandled_Expect_MessageAcknowledged() {
+        when(message.getJMSMessageID()).thenReturn("messageId");
```
[the remark this answers is not in the document]
Comment (mattd-kainos) [reply service/src/test/java/uk/nhs/adaptors/gp2gp/common/task/TaskConsumerTest.java:32]: deleted the when clauses instead, they were not needed

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/request/EhrExtractRequestHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class EhrExtractRequestHandlerTest {
+
+    public static final String REQUEST_ID = "041CA2AE-3EC6-4AC9-942F-0F6621CC0BFC";
+    public static final String CONVERSATION_ID = "DFF5321C-C6EA-468E-BBC2-B0E48000E071";
+    public static final String NHS_NUMBER = "9692294935";
+    private static final String TASK_ID = "3a93dfdd-5e72-4f23-8311-9f22772787af";
```
[the remark this answers is not in the document]
Comment (mattd-kainos) [reply service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/request/EhrExtractRequestHandlerTest.java:46]: These values are for the fixtures (xml files) used by this test specifically. EhrStatusConstants could be anything even if they just so happen to be the same right now.

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/request/EhrExtractRequestHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+            timestamp,
+            CONVERSATION_ID,
+            new EhrExtractStatus.EhrRequest(REQUEST_ID,
+                NHS_NUMBER,
+                "N82668-820670",
+                "B86041-822103",
+                "200000000205",
+                "200000001161",
+                "N82668",
+                "B86041")
```
[the remark this answers is not in the document]
Comment (mattd-kainos) [reply service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/request/EhrExtractRequestHandlerTest.java:109]: used a builder instead which is the better way to construct large objects like this

**Review by Broom94 — APPROVED** [review]

**Review by ElliotB-96 — APPROVED** [review]
Code looks good

**Pull request merged**